### PR TITLE
add event interval from rest client metric broken down by source

### PR DESCRIPTION
**Pull request by tkashem**

### CODE REVIEW

File: pkg/monitor/clientview/gatherer.go [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+}
+
+type gatherer struct {
+	start  time.Time
+	client prometheusv1.API
+}
+
+func (g *gatherer) query(ctx context.Context) (prometheustypes.Value, error) {
+	// TODO: should we include 5xx? "429|5..|<error>"
+	query := `rest_client_requests_total{code =~ "429|5..|<error>"}`
```
Comment (vrutkovs) [inline pkg/monitor/clientview/gatherer.go:57]: I wonder if we can add DNS resolve error - or it won't be recorded?

Comment (tkashem) [reply pkg/monitor/clientview/gatherer.go:57]: yes, when Michal's upstream PR merges, we can incorporate it.

**Comment by dinhxuanvu** [comment]
The upstream PR: https://github.com/kubernetes/kubernetes/pull/115357

File: pkg/monitor/clientview/gatherer.go (hunk 1 above)
Comment (tkashem) [reply pkg/monitor/clientview/gatherer.go:65]: to be more clear, if dns error happens code=<error>, any other network error, code=<error> also. (we can't make the distinction today with rest_client_requests_total. Michal's PR adds a new metric just to capture dns error.

**Comment by dinhxuanvu** [comment]
@p0lyn0mial @dgrisonnet Would like to have your inputs on this PR. Abu indicates this PR is ready for merge. We can either merge it and I can create subsequent PRs to address feedbacks or incorporate those changes into linked Vadim's PR.

**Comment by dgrisonnet** [comment]
I think we should close this one and wrap up the work on the PR Vadim opened and improved.

**Pull request closed without merging**